resspect/lightcurves_utils.py

```diff
@@ -0,0 +1,160 @@
+"""
+    utils for fit_lightcurves methods
+"""
+
+import io
+import tarfile
+from typing import AnyStr
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+
+from resspect.snana_fits_to_pd import read_fits
+
+SNPCC_LC_MAPPINGS = {
+    "snii": ['2', '3', '4', '12', '15', '17', '19', '20', '21', '24', '25',
+             '26', '27', '30', '31', '32', '33', '34', '35', '36', '37', '38',
+             '39', '40', '41', '42', '43', '44'],
+    "snibc": ['1', '5', '6', '7', '8', '9', '10', '11', '13', '14', '16',
+              '18', '22', '23', '29', '45', '28']
+}
+
+
+def read_file(file_path: str) -> list:
+    with open(file_path, "r") as file:
+        lines = [line.split() for line in file.readlines()]
+        return [line for line in lines if len(line) > 1]
+
+
+def get_snpcc_sntype(value: str) -> str:
+    if value in SNPCC_LC_MAPPINGS["snibc"]:
+        return 'Ibc'
+    elif value in SNPCC_LC_MAPPINGS["snii"]:
+        return 'II'
+    elif value == '0':
+        return 'Ia'
+    raise ValueError('Unknown supernova type!')
+
+
+def read_tar_file(file_path: str) -> AnyStr:
+    with tarfile.open(file_path, 'r:gz') as tar:
+        tar_members = tar.getmembers()[0]
+        return tar.extractfile(tar_members).read()
+
+
+def read_resspect_full_photometry_data(file_path: str) -> pd.DataFrame:
+    if '.tar.gz' in file_path:
+        tar_content = read_tar_file(file_path)
+        return pd.read_csv(io.BytesIO(tar_content))
+    elif '.FITS' in file_path:
+        _, full_photometry = read_fits(
+            file_path, drop_separators=True)
+        return full_photometry
+    else:
+        return pd.read_csv(file_path, index_col=False)
```

Comment:
put this in an `elif` and raise an exception if it's not a tar.gz, fits, or csv?